src/net.cpp

```diff
@@ -1640,7 +1642,7 @@ void CConnman::ThreadDNSAddressSeed()
                     {
                         LOCK(cs_vNodes);
                         for (const CNode* pnode : vNodes) {
-                            nRelevant += pnode->fSuccessfullyConnected && !pnode->fFeeler && !pnode->fOneShot && !pnode->m_manual_connection && !pnode->fInbound;
+                            nRelevant += pnode->fSuccessfullyConnected && (pnode->m_conn_type == ConnectionType::OUTBOUND || pnode->m_conn_type == ConnectionType::BLOCK_RELAY);
```

Comment:
ah yeah. totally agree with the nice compiler guarantee of switch statements. I've extracted `IsOutboundOrBlockRelayConn` into its own function, but looks like I missed incorporating the switch statement. I've updated locally and will push with next round of updates. 

one question- in the other property-test-with-switch-statement functions [example](https://github.com/bitcoin/bitcoin/pull/19316/files#diff-1a8b9d1ad0a6fda5e751286c73102fc2R814), I've had to add a default case because otherwise the build with `-Werror` enabled fails. is this a good way to handle? is there a cleaner way?

Reply:
updated

Reply:
I was going to suggest that this can be replaced with something like `if (pnode->IsOutboundOrBlockRelay())` - in which case the caller looks short and clean and the `switch` or chain of `if`s and the `enum` are hidden into the body of that method.

See https://github.com/bitcoin/bitcoin/pull/19316/files#r459397380

Reply:
> this can be replaced with something like if (pnode->IsOutboundOrBlockRelay())

yup, already done [link](https://github.com/bitcoin/bitcoin/pull/19316/files#diff-9a82240fe7dfe86564178691cc57f2f1R1651) :)